### DISCUSSION TITLE
Webview2 DragDrop/DragEnter Events

### DISCUSSION THREAD

**Discussion by senstar-mhuang** [MicrosoftEdge/WebView2Feedback#3973] [Q&A]
Hi, 

I have a .NET C# Webview2 control that's been properly initialized and showing a webpage, whose parent is a custom control. In our application, there are draggable objects that we handle with the Forms.Control.DragDrop and DragEnter events. 

` this.m_webBrowser.DragDrop += new System.Windows.Forms.DragEventHandler(this.BrowserPanel_DragDrop);
            this.m_webBrowser.DragEnter += new System.Windows.Forms.DragEventHandler(this.BrowserPanel_DragEnter)`

These events fire correctly in our other custom controls that inherit the Control class but they don't seem to fire when the same objects are dragged over the Webview2 control. Is this by design, i.e does Webivew2 have its own drag handling and/or suppresses these events manually? If so, is there a way to disable this behavior (as I would like to implement custom drag/drop events)? 

Thanks for your help.

**Comment by johna-ms** [comment]
Hi senstar-mhuang. WebView2 does not currently support Drag and Drop, though it is being worked on. Unfortunately no general workarounds come to mind. We are not suppressing drag/drop handler internally so we can't undo that for you. What are you trying to accomplish with the custom drag/drop handler?

**Comment by RatheeshVA** [comment]
Hi johna-ms, Is there any support for Drag and Drop feature of image files in HTML page which is shown in WebView2.
If so can you please share any reference. I couldn't find it anywhere.

**Comment by johna-ms** [comment]
@RatheeshVA this should only be broken for WinUI platforms. Support for that is being worked on as I mentioned above

**Comment by RatheeshVA** [comment]
@johna-ms I am having one more doubt.
Using WebBrowser, previously I could able to invoke JS script functions by defining those inside '_onclick_', '_ondblclick_' events inside '**img**' tag in HTML.
Eg:
**myWebpage.html** - Here I called MouseImageClick() in onclick event inside img tag
```
<script type="text/javascript" src="myScript.js"/>
<td>
<img src="filename1.png" onclick="MouseImageClick(filename1.png)" ondblclick="DblImageClick(filename1.png)" onmousedown="MouseDownImage(filename1.png)" />
</td>
```
**myScript.js** - I defined functions to drag image file from one source to dest on Left mouse button click.
```
function MouseImageClick(fileName)
{
    var evt = window.event;
    if( evt.button == ButtonLeft )
    {
        OBRViewerOCX.MoveImage(src, dest);
    }
    else if( evt.button == ButtonRight )
    {
        OBRViewerOCX.ShowImageMenu();
    }
}
```

But in WebView2, currently I could not able to do so by defining the functions in onclick or ondblclick in html page.
Can you please clarify my doubt.
